src/core/toga/widgets/table.py

```diff
@@ -175,11 +175,13 @@ def data(self, data):
         if data is not None:
             self._data.add_listener(self)
 
+        self._impl.refresh()
+
     @property
     def on_select(self):
         """ The callback function that is invoked when a row of the table is selected.
-        The provided callback function has to accept two arguments table (``:obj:Table`)
-        and row (``int`` or ``None``).
+        The provided callback function has to accept one argument, table
+        (``:obj:Table``), and keyword arguments (i.e. **kwargs).
```

Comment:
Sorry this was a misunderstanding. The user should provide a function that takes two positional arguments. 
We as Toga developer have to give the handler function `row` as a kwarg. Like you correctly did in the gtk.table file in line 50. So please revert this part.